File: server/my-family-ws/src/main/java/com/bytepushers/family/model/Permission.java
package com.bytepushers.family.model;

import jakarta.persistence.*;

@Entity
@Table(name = "permissions")
public class Permission {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String name;

    public Permission(){

    }

    public Permission(Integer id, String name) {
        this.id = id;
        this.name = name;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Permission{" +
                "id=" + id +
                ", name='" + name + '\'' +
                '}';
    }
}

File: client/my-family-app/src/app/models/permission.ts
export class Permission {
    #name: string;

    constructor(name: string) {
        this.#name = name;
    }

    public getName(): string {
        return this.#name;
    }

    public get name(): string {
        return this.#name;
    }

    public toString(): string {
        return `"permission": {
              "name": "${this.#name}"
            }`;
    }
}
